### PR TITLE
Issue 8943: Only check for too old postings for feeds

### LINKED ISSUE #8943: Post expiration interval can override user search query
### Demonstrated behavior

From my Friendica instance, if a user searches for a specific URL to grab a post from another system (I tested with https://yo.isurf.ca/notice/9wFh8IegUT7YpcAXqq ), but the post itself is older than an expiration setting (I believe the only expirations that this post is older than is the "Lifespan of remote items" and "Lifespan of unclaimed items"), then Friendica errors out and tells the user "No results".  The logs shows: 
`2020-07-28T20:36:58Z index [NOTICE]: Item created before expiration interval. {"created":"2020-06-19T20:38:21-07:00","expired":"2020-06-28T13:36:58-07:00", ....`

### Expected behavior

If a user goes to pull in a post from somewhere else in this fashion, it should override system expiration settings.

### Steps to reproduce the problem

Have the two settings mentioned set to under 30 days and then search for a post than is older days.

### Friendica version you encountered the problem

2020.07 stable

### Friendica source (git, zip)

git

### PHP version

7.4.8

### SQL version

MariaDB 10.3.23

### PR DESCRIPTION
**Pull request by annando**
This fixes https://github.com/friendica/friendica/issues/8943

I just checked the code. We expire based on the date the message arrived at the system. So no situation should occur when an item would be fetched and directly being expired.

This check does make sense for feeds, so it is now only used there.

### CODE REVIEW

File: src/Model/Item.php [hunk 1]
```diff
@@ -1446,6 +1425,38 @@ public static function isValid(array $item)
 		return true;
 	}
 
+	/**
+	 * Check if the item array is too old
+	 *
+	 * @param array $item
+	 * @return boolean item is too old
+	 */
+	public static function tooOld(array $item)
```
Comment (MrPetovan) [inline src/Model/Item.php]: Please rename to `isTooOld` for consistency with `isValid`.

File: src/Protocol/Feed.php [hunk 2]
```diff
@@ -503,7 +503,10 @@ public static function import($xml, array $importer = [], array $contact = [])
 				$items[] = $item;
 				break;
 			} elseif (!Item::isValid($item)) {
-				Logger::info('Feed is invalid', ['created' => $item['created'], 'uid' => $item['uid'], 'uri' => $item['uri']]);
+				Logger::info('Feed item is invalid', ['created' => $item['created'], 'uid' => $item['uid'], 'uri' => $item['uri']]);
+				continue;
+			} elseif (Item::tooOld($item)) {
```
Comment (MrPetovan) [inline src/Protocol/Feed.php]: If this is only used here, please move the method to this file.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Protocol/Feed.php (hunk 2 above)
Comment (annando) [reply src/Protocol/Feed.php]: I see it equal to the `isValid` function. So I think it does belong there. And possibly we want to use it for some other places in the future as well, like for example importing the outboxes of AP accounts.

File: src/Model/Item.php (hunk 1 above)
Comment (annando) [reply src/Model/Item.php]: Renamed

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**